src/nixpacks/app.rs

```diff
@@ -32,6 +35,22 @@ impl App {
         Ok(contents)
     }
 
+    pub fn find_match(&self, re: &Regex, pattern: &str) -> bool {
+        let full_pattern = self.source.join(pattern);
+        let entries = glob(full_pattern.to_str().unwrap()).expect("Failed to read glob pattern");
+
+        let mut matched = false;
+        for entry in entries {
+            let path_buf = fs::canonicalize(entry.unwrap()).unwrap();
+            let f = self.read_file(path_buf.to_str().unwrap()).unwrap();
+            if re.find(f.as_str()).is_some() {
+                matched = true;
+                break;
+            }
+        }
+        matched
+    }
```

Comment:
Instead of unwrapping here I think the the `find_match` function should return a `Result`. Then you can `?` instead of unwrap the function calls.

Reply:
For consistency, I think it should keep it as a bool because things like `includes_file` do it like this.

Reply:
Then we should make `includes_file` return a Result. A Result is much more useful since we can handle the error further up. Most other places in the providers/builder deal with results.

Reply:
There isn't actually a point to making `includes_file` return a `Result` because we manually check if it is ok with`is_ok`.

Reply:
Sure, but your point above was about consistency. We shouldn't unwrap here and if we want everything to be consistent then we can return a Result everywhere. Not the other way around.

However, I think it is fine to sometimes return a Result and sometimes not. But unwrapping is akin to `!` in TypeScript, which is to be avoided.

Reply:
Okay, I agree with you. However, `includes_file` shouldn't. We would be hardcoding an Ok() in `includes_file`.